### PR TITLE
Migrate access history modal to functional component

### PR DESCRIPTION
**Pull request by larkox**
#### Summary
Migrate access history modal to functional component

#### Ticket Link
None

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: webapp/channels/src/components/access_history_modal/access_history_modal.test.tsx [hunk 1]
```diff
@@ -38,26 +40,23 @@ describe('components/AccessHistoryModal', () => {
         expect(wrapper.find(AuditTable).exists()).toBe(true);
     });
 
-    test('should have called actions.getUserAudits when onShow is called', () => {
+    test('should have called actions.getUserAudits only when first rendered', () => {
```
Comment (larkox) [inline webapp/channels/src/components/access_history_modal/access_history_modal.test.tsx:43]: Some tests descriptions may no longer apply, so we do our best to test the same and update the description accordingly.

File: webapp/channels/src/components/access_history_modal/access_history_modal.test.tsx [hunk 2]
```diff
@@ -38,26 +40,23 @@ describe('components/AccessHistoryModal', () => {
...
         const actions = {
             getUserAudits: jest.fn(),
         };
         const props = {...baseProps, actions};
-        const wrapper = shallow<AccessHistoryModal>(
-            <AccessHistoryModal {...props}/>,
-        );
+        const view = render(withIntl(<AccessHistoryModal {...props}/>));
```
Comment (larkox) [inline webapp/channels/src/components/access_history_modal/access_history_modal.test.tsx:48]: `shallow` does not run the effects, so we move on with `render` from the `testing-library/react`.

File: webapp/channels/src/components/access_history_modal/access_history_modal.test.tsx [hunk 3]
```diff
@@ -38,26 +40,23 @@ describe('components/AccessHistoryModal', () => {
...
-        );
-
-        wrapper.setState({show: true});
-        wrapper.instance().onHide();
-        expect(wrapper.state('show')).toEqual(false);
+    test('should hide', async () => {
+        render(withIntl(<AccessHistoryModal {...baseProps}/>));
+        await waitFor(() => screen.getByText('Access History'));
+        fireEvent.click(screen.getByLabelText('Close'));
+        await waitForElementToBeRemoved(() => screen.getByText('Access History'));
```
Comment (larkox) [inline webapp/channels/src/components/access_history_modal/access_history_modal.test.tsx:60]: The previous test was testing internal state, which is not the best practice. Here we are testing with internals of the children component, which is also not the best practice. Either way, both test the same behavior.

**Review by M-ZubairAhmed — APPROVED** [review]

**Review by jespino — APPROVED** [review]
LGTM

**Comment by larkox** [comment]
@yasserfaraazkhan QA Notes:

This touches the "Access History Modal". This is the modal that appears when you click `View Access history` in the security section of the user profile settings.

No changes in behavior are expected.

**Review by yasserfaraazkhan — APPROVED** [review]
working as expected and all e2e tests have passed

**Pull request merged**